src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -4,167 +4,74 @@
 
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Diagnostics;
-using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
-using Microsoft.Win32.SafeHandles;
+using System.Threading;
 
 namespace Internal.Cryptography.Pal
 {
     internal class CollectionBackedStoreProvider : IStorePal
     {
-        private readonly X509Certificate2[] _certs;
-        private static readonly SafeEvpPKeyHandle InvalidPKeyHandle = new SafeEvpPKeyHandle(IntPtr.Zero, false);
+        private readonly bool _cloneOnRead;
```

Comment:
Yes, splitting the interfaces is the best approach. I just figured that suggestion would be a non-starter due to adding a huge amount of work to an already big PR.

But if you’re willing (either here or in a separate PR), I’ll be delighted with that approach ☺